internal/cmd/connect/command_plugin_install.go

```diff
@@ -0,0 +1,461 @@
+	}
+	calculatedSha1Checksum := fmt.Sprintf("%x", sha1.Sum(archive))
+	if calculatedSha1Checksum != pluginManifest.Archive.Sha1 {
+		return errors.Errorf(checksumErrorMsg, "SHA1", calculatedSha1Checksum, pluginManifest.Archive.Sha1, pluginManifest.Name)
+	}
+
+	zipReader, err := zip.NewReader(bytes.NewReader(archive), int64(len(archive)))
+	if err != nil {
+		return errors.Wrapf(err, "failed to open remote archive file %s", archive)
+	}
+
+	return unzipPlugin(pluginManifest, zipReader.File, pluginDir)
+}
+
+func unzipPlugin(pluginManifest *cpstructs.Manifest, zipFiles []*zip.File, pluginDir string) error {
+	relativeInstallationDir := filepath.Join(pluginDir, fmt.Sprintf("%s-%s", pluginManifest.Owner.Username, pluginManifest.Name))
+	installationDir, err := filepath.Abs(relativeInstallationDir)
+	if err != nil {
+		return errors.Wrapf(err, "failed to resolve absolute path for directory %s", relativeInstallationDir)
+	}
+
+	for _, zipFile := range zipFiles {
+		versionPrefix := fmt.Sprintf("%s-%s-%s", pluginManifest.Owner.Username, pluginManifest.Name, pluginManifest.Version)
+		destFilePath := filepath.Join(installationDir, strings.TrimPrefix(zipFile.Name, versionPrefix))
+
+		createDirectoryErrorMsg := "failed to create directory %s on local storage"
+		if zipFile.FileInfo().IsDir() {
+			if err := os.MkdirAll(destFilePath, 0755); err != nil {
+				return errors.Wrapf(err, createDirectoryErrorMsg, destFilePath)
+			}
+			continue
+		} else {
+			if err := os.MkdirAll(filepath.Dir(destFilePath), 0755); err != nil {
+				return errors.Wrapf(err, createDirectoryErrorMsg, filepath.Dir(destFilePath))
+			}
+		}
+
+		zipFileReader, err := zipFile.Open()
+		if err != nil {
+			return errors.Wrapf(err, "failed to read file %s from archive", zipFile.Name)
+		}
+		defer zipFileReader.Close()
+
+		destFile, err := os.OpenFile(destFilePath, os.O_CREATE|os.O_RDWR, 0644)
+		if err != nil {
+			return err
+		}
+		defer destFile.Close()
+
+		if _, err := io.Copy(destFile, zipFileReader); err != nil {
+			return errors.Wrapf(err, "failed to copy file %s from archive to local file %s", zipFile.Name, destFilePath)
+		}
+	}
+
+	return nil
+}
+
+func checkLicenseAcceptance(pluginManifest *cpstructs.Manifest, prompt form.Prompt, force bool) error {
+	for _, license := range pluginManifest.Licenses {
+		if force {
+			output.Printf("Implicitly agreeing to the following license: %s (%s)\n", license.Name, license.Url)
+		} else {
+			f := form.New(form.Field{
+				ID:        "confirm",
+				Prompt:    fmt.Sprintf("License: %s (%s)\nI agree to this software license agreement.", license.Name, license.Url),
+				IsYesOrNo: true,
+			})
+			if err := f.Prompt(prompt); err != nil {
+				return err
+			}
+			if !f.Responses["confirm"].(bool) {
+				return errors.New("you must accept all license agreements to install this plugin")
+			}
+		}
+	}
+	output.Print("\n")
+
+	return nil
+}
+
+func addDryRunPrefix(msg string) string {
```

Comment:
I wonder if other commands could reuse this?

Reply:
Yup. I moved this to utils and changed the two other instances of dry run messages to call this.